src/extremeweatherbench/data_loader.py

```diff
@@ -0,0 +1,161 @@
+logger.setLevel(logging.INFO)
+
+
+def open_forecast_dataset(
+    eval_config: config.Config,
+    forecast_schema_config: config.ForecastSchemaConfig,
+) -> xr.Dataset:
+    """Open the forecast dataset specified for evaluation.
+
+    Args:
+        eval_config: The evaluation configuration.
+        forecast_schema_config: The forecast schema configuration.
+
+    Returns:
+        The opened forecast dataset.
+    """
+    logger.debug("Opening forecast dataset")
+    if "://" in eval_config.forecast_dir:
+        filesystem = eval_config.forecast_dir.split("://")[0]
+    else:
+        filesystem = "file"
+    fs = fsspec.filesystem(filesystem)
+    file_list = fs.ls(eval_config.forecast_dir)
+    file_types = set([file.split(".")[-1] for file in file_list])
+
+    if len(file_types) > 1:
+        if (
+            "parq" not in eval_config.forecast_dir
+            and "zarr" not in eval_config.forecast_dir
+        ):
+            raise TypeError("Multiple file types found in forecast path.")
+        if "parq" in eval_config.forecast_dir:
+            # kerchunked parq refs seem to consistently need two "compute" calls
+            # the first one doesn't actually load to memory
+            forecast_ds = open_kerchunk_reference(
+                eval_config.forecast_dir, forecast_schema_config
+            )
+        elif "zarr" in eval_config.forecast_dir:
+            forecast_ds = xr.open_zarr(eval_config.forecast_dir, chunks="auto")
+    elif len(file_types) == 1:
+        if "zarr" in file_types:
+            forecast_ds = xr.open_zarr(file_list, chunks="auto")
+        elif "json" in file_types:
+            forecast_ds = open_kerchunk_reference(file_list[0], forecast_schema_config)
+        else:
+            raise TypeError(
+                "Unknown file type found in forecast path, only json, parquet, and zarr are supported."
+            )
+    else:
+        raise FileNotFoundError("No files found in forecast path.")
+
+    mapping = {
+        getattr(forecast_schema_config, field.name): field.name
+        for field in dataclasses.fields(forecast_schema_config)
+    }
+
+    filtered_mapping_data_vars = {
+        old: new for old, new in mapping.items() if old in forecast_ds.data_vars
+    }
+    filtered_mapping_coords = {
+        old: new for old, new in mapping.items() if old in forecast_ds.coords
+    }
+    filtered_mapping = {**filtered_mapping_data_vars, **filtered_mapping_coords}
+    forecast_ds = forecast_ds.rename(filtered_mapping)
+    forecast_ds = utils.convert_dataset_lead_time_to_int(forecast_ds)
+
+    return forecast_ds
+
+
+def open_kerchunk_reference(
+    file,
+    forecast_schema_config: config.ForecastSchemaConfig,
+    remote_protocol: str = "s3",
+) -> xr.Dataset:
+    """Open a dataset from a kerchunked reference file in parquet or json format.
+    This has been built for the CIRA MLWP S3 bucket's data, but can work with other
+    data as well if it is 6 hourly.
+
```

Comment:
Why does it matter if the data is 6-hourly or not? In your explanation of the assumptions here, you should specify what you are expecting, and then you can note the CIRA MLWP bucket (link to the AWS Public Datasets page for it (https://registry.opendata.aws/aiwp/)) satisfies this convention.